src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -52,36 +52,25 @@ const Root: React.FC<SegmentedControlProps> = ({
     }),
     sxProp as SxProp
   )
-  const focusInStrategy: FocusZoneHookSettings['focusInStrategy'] = useCallback(() => {
-    if (segmentedControlContainerRef.current) {
-      const buttons = Array.from(segmentedControlContainerRef.current.querySelectorAll('button'))
-      return buttons[selectedIndex]
-    }
-  }, [segmentedControlContainerRef, selectedIndex])
 
-  // TODO: eliminate the need for a custom `getNextFocusable`
-  // We need a custom `getNextFocusable` because after selecting a button, `useFocusZone`
-  // does not correctly move focus to the next or previous button.
-  // This doesn't happen if we don't pass any dependencies to `useFocusZone`, but we need to
-  // pass `focusInStrategy` as a dependency so that the selected button gets focus first.
-  const getNextFocusable: FocusZoneHookSettings['getNextFocusable'] = (direction, from) => {
+  const focusInStrategy: FocusZoneHookSettings['focusInStrategy'] = () => {
     if (segmentedControlContainerRef.current) {
-      const buttons = Array.from(segmentedControlContainerRef.current.querySelectorAll('button'))
-      const nextIndex = buttons.indexOf(from as HTMLButtonElement) + (direction === 'previous' ? -1 : 1)
+      // we need to use type assertion because querySelector returns "Element", not "HTMLElement"
+      type SelectedButton = HTMLButtonElement | undefined
 
-      return buttons[nextIndex]
+      const selectedButton = segmentedControlContainerRef.current.querySelector(
+        'button[aria-current="true"]'
```

Comment:
@mperrotti Is this a safe assumption to make? 

By using `aria-current` instead of `selectedIndex`, we are able to remove dependency array and useCallback which simplified the implementation :)

Reply:
Yup!

Reply:
perfect!